pkg/model/crd.go

```diff
@@ -732,14 +734,14 @@ func (r *CRD) GoCodeSetInput(
 		sourceAdaptedVarName := sourceVarName
 		crdField, found = r.SpecFields[renamedName]
 		if found {
-			sourceAdaptedVarName += ".Spec"
+			sourceAdaptedVarName += r.StringStore[SpecFieldPath]
```

Comment:
Instead of the `map[string]string` here, if you went with the `PrefixConfig` struct I suggest above, you could have little helper methods on `CRD` like so:

```go

func (crd *CRD) specFieldPrefix() string {
    if crd.cfg == nil || crd.cfg.Prefix == nil {
        return ".Spec"
    }
    return c.Prefix.SpecField
}

func (crd *CRD) statusFieldPrefix() string {
    if crd.cfg == nil || crd.cfg.Prefix == nil {
        return ".Status"
    }
    return c.Prefix.StatusField
}
```

Reply:
I was thinking that for both ack and crossplane, we'd have a hard-coded default `Config` object and use the `generator.yaml` as an overlay on top of that `Config` so we'd make sure there is always a value in those fields.